### ISSUE TITLE
[Bug?]: Can't use `Client` as a model name and generate scaffolds for it

### ISSUE THREAD

**Issue by esteban-url** [redwoodjs/redwood#6946]
### What's not working?

If you create a model with the name `Client` and then generate the scaffold, your app will break when you navigate to the show page of a client, say http://localhost:8910/clients/1

the variable `client` seem to get over written by `apolloClient`.

notice the `console.log` in line 75 

![image](https://user-images.githubusercontent.com/14810250/203473506-a90bf0f8-2437-44cd-82a2-5aa245a4027e.png)

it throws this error (notice the console)

![image](https://user-images.githubusercontent.com/14810250/203470137-83e8aa6c-6372-4bbe-8e49-afc29235de7f.png)

### How do we reproduce the bug?

1. create a new Redwood project 
2. add a model named `Client`
3. create migration and generate scaffold
4. run the server
5. create a client via the ui. 
6. From the clients list click on the show button
7. 😫

theres a repo that reproduces the error here: https://github.com/esteban-url/rw-client-model-bug

### What's your environment? (If it applies)

```shell
System:
    OS: macOS 13.0.1
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.18.1 - /private/var/folders/hd/mgflxzlj64vff4rm6w4rjw840000gn/T/xfs-888881df/node
    Yarn: 3.2.3 - /private/var/folders/hd/mgflxzlj64vff4rm6w4rjw840000gn/T/xfs-888881df/yarn
  Databases:
    SQLite: 3.39.4 - /usr/bin/sqlite3
  Browsers:
    Chrome: 107.0.5304.110
    Firefox: 97.0.1
    Safari: 16.1
  npmPackages:
    @redwoodjs/core: 3.4.0 => 3.4.0
```

### Are you interested in working on this?

- [X] I'm interested in working on this

**Comment by pantheredeye** [comment]
Would an easy stopgap be to make a note in the docs regarding this plus any other naming conflict? I am pretty sure I have run across another conflict before. I searched for it, but I cannot find it.

**Comment by esteban-url** [comment]
Yeah I think the best escenario is create another issue to add the info to the docs with info about all the known "reserved" or "restricted" words, and try to solve them one by one in independent gh issues. thoughts?

**Comment by Josh-Walker-GM** [comment]
I did some very quick investigating with your repo. 

I tried changing the query name in the other cell (`web/src/components/Control/ControlCell/ControlCell.tsx`) and naming that query instead of control. So changing it to:
```gql
query FindControlById($id: Int!) {
  client: control(id: $id) {
    id
    name
    description
    type
    value
    createdAt
    updatedAt
  }
}
```
When you then visit the control pages they break in the same way as you report the client ones too. So I guess any graphql query with "client" like that breaks.

I then looked at changing the success export of the cell from:
```ts
export const Success = ({client}: CellSuccessProps<FindClientById>) => {
```
to
```ts
export const Success = (stuff: CellSuccessProps<FindClientById>) => {
```
and console logged `stuff`. We see:

![image](https://user-images.githubusercontent.com/56300765/205154727-f83e3f26-d486-4275-b178-366c25961928.png)

where I have renamed the actual graphql query to form `client: ` to `THE_QUERY: `. I guess `CellSuccessProps` within `@redwoodjs/web` is a good place to start looking but I am really just guessing off of what I've seen here and might be totally wrong.

**Comment by Josh-Walker-GM** [comment]
Also: The docs for all the returned values from `useQuery` are [here](https://www.apollographql.com/docs/react/data/queries/#result). The `client` is listed under the network info subsection.

**Comment by esteban-url** [comment]
this is stellar work @Josh-Walker-GM thank you!! I'll try it out, see if anything else breaks!

**Comment by esteban-url** [comment]
I've created a very simple PR (#7024) that completely solves the problem and doesn't seem to cause any other issues.

Thanks @Josh-Walker-GM you really served this on a silver plater!

**Comment by michaelmcandrew** [comment]
Hey @esteban-url - thanks for your work on this!

It looks like the fix is passing the tests. Is there anything stopping it from being merged?

**Comment by esteban-url** [comment]
@michaelmcandrew hi! since this is a breaking change it does required more consideration and scrutiny from the core team members. I've received multiple feedback  and change requests, I think the PR is complete but it will likely be merged as they get closer to cutting the 5.0 release. 

Is this stopping you in any way? have you cloned the branch and tested it, maybe your insights can help us know if it requieres any more adjustments. 

In any case, maybe @jtoar can better answer your question.

**Comment by michaelmcandrew** [comment]
Thanks for your reply, @esteban-url, and your insights into the release process :)

For now we are just using a different word (Accounts). We'd like to switch to Clients since the terminology is a much better fit but I totally understand that this is a breaking change and that we might have to wait before we can do that.

> have you cloned the branch and tested it

We're just getting started on our Redwood journey and we're keen to keep dev on the stable release so might skip the testing unless you think that would make a difference.

In any case, thanks for your time and for the fix - it is appreciated.

**Comment by esteban-url** [comment]
you are welcome! I'm on the same boat, would like to use `client`, but i'm "stuck" with `customer` for the time being.

You are good staying on the stable release. Personally I only go off it in dev branches to test out release candidates or a specific feature such as this one. Hopefully we can get this in 5.0! 🤞

**Comment by jtoar** [comment]
@esteban-url thanks for your patience, it's queued up for v5, we'll focus on it this week.

**Comment by esteban-url** [comment]
great news! thanks @jtoar! let me know if there is anything else I can do

**Comment by michaelmcandrew** [comment]
Thanks for this!
